base/linalg/dense.jl

```diff
@@ -589,11 +594,13 @@ julia> log(A)
 """
 function log(A::StridedMatrix{T}) where T
     # If possible, use diagonalization
-    if issymmetric(A) && T <: Real
-        return log(Symmetric(A))
+    if T <: Real
```

Comment:
Perhaps keep this on the same line, but the order change is good.
```julia
if T <: Real && issymmetric(A)
    # ...
end
```

Reply:
As commented in the other PR, the point here is to avoid checking whether a real matrix is symmetric twice.